src/Model/ROI.py

```diff
@@ -639,12 +641,23 @@ def calc_roi_polygon(curr_roi, curr_slice, dict_rois_contours, pixmap_aspect=1):
 
     list_polygons = []
     pixel_list = dict_rois_contours[curr_roi][curr_slice]
+    dataset = PatientDictContainer().dataset[0]
+
     for i in range(len(pixel_list)):
         list_qpoints = []
         contour = pixel_list[i]
         for point in contour:
-            curr_qpoint = QtCore.QPoint(point[0], point[1] * pixmap_aspect)
-            list_qpoints.append(curr_qpoint)
+            if dataset['Rows'].value != DEFAULT_WINDOW_SIZE:
```

Comment:
I think it would be even better if you can move this if else statement outside of the for loop because the dataset['Rows'].value is the same for all point in contour. So it is a bit redundant and time-consuming if the program have to check this for every point in the contour.

Reply:
Made a new variable to store the Boolean result, checks the variable rather than dataset['rows']